src/Geno/C++/GUI/Widgets/TextEdit.cpp

```diff
@@ -3439,3 +3507,590 @@ std::vector< int > TextEdit::CursorsNotInText( File& rFile )
 
 	return Cursors;
 } // CursorsNotInText
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::ClearSearch( File& rFile )
+{
+	delete rFile.SearchDiag.SearchResult;
+	rFile.SearchDiag.SearchResult = nullptr;
+
+	StatusBar::Instance().SetSearchResultInfo("", 0);
+} // ClearSearch
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::PrepareSearchString( std::string& rSearchString )
+{
+	for( int i = 0; i < ( int )rSearchString.length(); i++ )
+	{
+		char C = rSearchString[ i ];
+
+		if( C == '\\' )
+		{
+			if( i == ( int )rSearchString.length() - 1 ) break;
+
+			char C2 = rSearchString[ i + 1 ];
+
+			switch( C2 )
+			{
+				case 'n':
+					rSearchString[ i ] = '\n';
+					break;
+				case 't':
+					rSearchString[ i ] = '\t';
+					break;
+				default:
+					continue;
+			}
+
+			rSearchString.erase( rSearchString.begin() + i + 1 );
+		}
+	}
+} // PrepareSearchString
+
+//////////////////////////////////////////////////////////////////////////
+
+char toLower( char c )
+{
+	//TODO: Possibly handle other languages as well
+	if( c >= 'A' && c <= 'Z' )
+	{
+		return c + 32;
+	}
+
+	return c;
+}
```

Comment:
There's already https://en.cppreference.com/w/cpp/locale/tolower that supports other locales even!

Reply:
That's what I used in the beginning until I realized it was really slow. Like it totally kills performance.

Reply:
Wait! It only seems rubbish when you pass it the locale.

Reply:
Ah, yeah there seems to be another version that doesn't take a locale